/*
    Croc GUI. Additional tool for croc (https://github.com/schollz/croc):
    "Easily and securely send things from one computer to another"
    Copyright (C) 2022, Croc GUI team

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

#ifndef WIDGET_H
#define WIDGET_H

#include "socketchecker.h"

#include <QWidget>

QT_BEGIN_NAMESPACE
namespace Ui { class Widget; }
QT_END_NAMESPACE

class Widget : public QWidget
{
    Q_OBJECT

public:
    Widget(QWidget *parent = nullptr);
    ~Widget();

private:
    void initSendTab();
    void initReceiveTab();
    void initSettingsTab();
    void saveSettings();
    void initHistoryTab();

    Ui::Widget *ui;

private slots:
    void checkFilepathToSend();
    void historyMenuRequest(QPoint pos);

    void testCrocExec();
    void restoreCrocTestButton(QString ver);

    void runProxyTest();
    void restoreProxyTestButton(bool status);

    void runCustomRelayTest();
    void restoreCustomRelayTestButton(bool status);

signals:
    void crocTestFinished(QString ver);
    void proxyTestFinished(bool status);
    void customRelayTestFinished(bool status);
};
#endif // WIDGET_H
